cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/service/BusinessMetadataStore.java

```diff
@@ -167,6 +169,34 @@ public Void apply(BusinessMdsIterable input) throws Exception {
     });
   }
 
+  /**
+   * Search to the underlying Business Metadata Dataset.
+   */
+  public Iterable<BusinessMetadataRecord> searchMetadata(final String searchQuery) {
+    return searchMetadataOnType(searchQuery, MetadataSearchTargetType.ALL);
+  }
+
+  /**
+   * Search to the underlying Business Metadata Dataset for a target type.
+   */
+  public Iterable<BusinessMetadataRecord> searchMetadataOnType(final String searchQuery,
+                                                        final MetadataSearchTargetType type) {
+    return txnl.executeUnchecked(new TransactionExecutor.Function<BusinessMdsIterable,
+      Iterable<BusinessMetadataRecord>> () {
+      @Override
+      public Iterable<BusinessMetadataRecord> apply(BusinessMdsIterable input) throws Exception {
+        // Currently we support two types of search formats: value and key=value.
+        // Check for exiatance of '=' char to make sure we did search in the right indexed column.
+        if (searchQuery.contains("=")) {
```

Comment:
But the design doc said we use key:value format so let me change it to that for now.